src/shogun/io/openml/OpenMLRun.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#include <shogun/evaluation/CrossValidationStorage.h>
+#include <shogun/io/openml/OpenMLRun.h>
+#include <shogun/io/openml/ShogunOpenML.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/machine/Machine.h>
+
+using namespace shogun;
+
+std::shared_ptr<OpenMLRun> OpenMLRun::run_model_on_task(
+    std::shared_ptr<CSGObject> model, std::shared_ptr<OpenMLTask> task)
+{
+	SG_SNOTIMPLEMENTED
+	return std::shared_ptr<OpenMLRun>();
+}
+
+std::shared_ptr<OpenMLRun> OpenMLRun::run_flow_on_task(
+    std::shared_ptr<OpenMLFlow> flow, std::shared_ptr<OpenMLTask> task)
+{
+	auto data = task->get_dataset();
+	std::shared_ptr<CFeatures> features = nullptr;
+	std::shared_ptr<CLabels> labels = nullptr;
+
+	auto model = ShogunOpenML::flow_to_model(std::move(flow), true);
+
+	labels = data->get_labels();
+	features = data->get_features(data->get_default_target_attribute());
+
+	auto storage = std::make_shared<CrossValidationStorage>();
+
+	if (task->get_split()->contains_splits())
+	{
+		auto machine = std::dynamic_pointer_cast<CMachine>(model);
+		if (!machine)
+		{
+			SG_SERROR("INTERNAL ERROR: failed to cast model to machine!\n")
```

Comment:
please give the user more infos here. you could use `get_name()` to print the class name. No need to write INTERNAL_ERROR

Reply:
yup, this is just a debugging message for myself. this will be replaced with `->as<CMachine>` once the smart pointers are ready